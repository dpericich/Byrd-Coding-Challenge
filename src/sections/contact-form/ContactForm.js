import React from "react";
import ContactInput from "./components/contact-form-input/ContactInput";
import "./contact-form-section.css";
import logoMain from "../../assets/logo-main.png";
import Header from "../../components/Header";

class ContactForm extends React.Component {
  state = {
    name: "",
    email: "",
    message: "",
  };

  // Callback function to lift name state up from ContactInput
  updateName = (event) => {
    this.setState({ name: event.target.value });
  };

  // Callback function to lift email state up from ContactInput
  updateEmail = (event) => {
    this.setState({ email: event.target.value });
  };

  // Callback function to lift message state up from ContactInput
  updateMessage = (event) => {
    this.setState({ message: event.target.value });
  };

  // Callback to print alert of the form data gathered from the contact form
  confirmSubmitForm = (event) => {
    alert(
      `Hello, ${this.state.name} we got your message. Check at ${this.state.email} for a response.`
    );
    event.preventDefault();
  };

  render() {
    return (
      <div className="contact-form-section">
        <Header color="orange" />
        <ContactInput
          name={this.state.name}
          updateName={this.updateName}
          email={this.state.email}
          updateEmail={this.updateEmail}
          message={this.state.message}
          updateMessage={this.updateMessage}
          confirmSubmitForm={this.confirmSubmitForm}
        />
      </div>
    );
  }
}

export default ContactForm;
