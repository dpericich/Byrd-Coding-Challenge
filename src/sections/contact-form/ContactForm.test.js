import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import ContactForm from "./ContactForm";

describe("<ContactForm />", () => {
  test("it renders the contact form", () => {
    const component = shallow(<ContactForm />);
    expect(toJSON(component)).toMatchSnapshot();
  });

  // When DOM error styling is introduced for the inputs, I want to simulate different input
  // combinations off of the form submit to check that they return the correct alert values
});
