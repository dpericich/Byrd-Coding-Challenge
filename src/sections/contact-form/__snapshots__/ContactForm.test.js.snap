// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> it renders the contact form 1`] = `
<div
  className="contact-form-section"
>
  <Header
    color="orange"
  />
  <ContactInput
    confirmSubmitForm={[Function]}
    email=""
    message=""
    name=""
    updateEmail={[Function]}
    updateMessage={[Function]}
    updateName={[Function]}
  />
</div>
`;
